### PR TITLE
Update README.md and add some documents about development in docs/ tree.

### PR DESCRIPTION
**Pull request by tpiperatgod**
- Update ```README.md```
- Create directory ```docs\```
- Add ```OpenFunction-architecture.png```
- Add some instructions on the development workflow

### CODE REVIEW

**Comment by tpiperatgod** [comment]
Optimized the ```OpenFunction-architecture.png```, replaced user avatar

File: README.md [hunk 1]
```diff
@@ -1,16 +1,158 @@
...
+
+     In order to access your mirror repository, you need to create a secret. You can create this secret by editing the ``username`` and ``password`` fields in ``config/samples/registry-account.yaml``, and then apply it.
+
+     ```shell
+     kubectl apply -f config/samples/registry-account.yaml
+     ```
+
+  2. Creating functions
+
+     Here is a simple Function case for you, modify the ``spec.image`` field in ``config/samples/core_v1alpha1_function.yaml`` to your own repository address:
```
Comment (benjaminhuo) [inline README.md:79]: =>
For sample function below, modify the ``spec.image`` field in ``config/samples/core_v1alpha1_function.yaml`` to your own container registry address:

File: README.md [hunk 2]
```diff
@@ -1,16 +1,158 @@
...
+
+     Use the following command to create this Function:
+
+     ```shell
+     kubectl apply -f config/samples/core_v1alpha1_function.yaml
+     ```
+
+  3. Result observation
+
+     You can observe the process of Function with the following command:
```
Comment (benjaminhuo) [inline README.md:98]: =>
You can observe the process of a function with the following command:

File: README.md [hunk 3]
```diff
@@ -1,16 +1,158 @@
...
+
+     ```shell
+     kubectl get functions.core.openfunction.io
+     
+     NAME              AGE
+     function-sample   5s
+     ```
+
+     You can also observe the process of Builder in the [Tekton Dashboard](https://tekton.dev/docs/dashboard/).
```
Comment (benjaminhuo) [inline README.md:107]: a builder

File: README.md [hunk 4]
```diff
@@ -1,16 +1,158 @@
...
+
+     Finally, you can observe the final state of the case in the Serving:
```
Comment (benjaminhuo) [inline README.md:109]: case => function workload

File: README.md [hunk 5]
```diff
@@ -1,16 +1,158 @@
...
+
+     ```shell
+     kubectl get servings.core.openfunction.io
+     
+     NAME                      AGE
+     function-sample-serving   15s
+     ```
+
+     At this point, you can view the service entry for the case function with the following command:
```
Comment (benjaminhuo) [inline README.md:118]: =>
You can now find out the service entry of the function with the following command:

File: docs/development/development-guide.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Development Guide
+
+---
+
+This document walks you through how to get started with building KubeSphere in your local environment.
```
Comment (benjaminhuo) [inline docs/development/development-guide.md:5]: building KubeSphere
=>
building OpenFunction

File: docs/development/development-guide.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+See [quickstart](quickstart.md)
+
+## Development Workflow
+
+See [development-workflow](development-workflow.md)
+
+## Pull Requests
+
+See [pull-requests](https://github.com/kubesphere/community/blob/master/developer-guide/development/pull-requests.md)
```
Comment (benjaminhuo) [inline docs/development/development-guide.md:19]: it's better to put pull-requests.md in OpenFunction development docs:
https://github.com/kubesphere/community/blob/master/developer-guide/development/pull-requests.md

File: docs/development/development-workflow.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+---
+
+## Step 1. Fork
+
+1. Visit https://github.com/OpenFunction/OpenFunction
+2. Click `Fork` button to create a fork of the project to your GitHub account.
+
+## Step 2. Clone fork to local storage
+
+Per Go's [workspace instructions](https://golang.org/doc/code.html#Workspaces), place KubeSphere code on your `GOPATH` using the following cloning procedure.
```
Comment (benjaminhuo) [inline docs/development/development-workflow.md:12]: KubeSphere
=>
OpenFunction

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

**Pull request merged**